source/math/IntVectorList.ooc

```diff
@@ -0,0 +1,78 @@
+/*
+* Copyright (C) 2014 - Simon Mika <simon@mika.se>
+*
+* This sofware is free software; you can redistribute it and/or
+* modify it under the terms of the GNU Lesser General Public
+* License as published by the Free Software Foundation; either
+* version 2.1 of the License, or (at your option) any later version.
+*
+* This software is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+* Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with this software. If not, see <http://www.gnu.org/licenses/>.
+*/
+use ooc-collections
+import math
+import FloatVectorList
+
+IntVectorList: class extends VectorList<Int> {
+	init: func ~default {
+		super()
+	}
+	init: func ~heap (capacity: Int) {
+		super(capacity)
+	}
+	init: func ~fromVectorList (other: VectorList<Int>) {
+		super(other _vector)
+		this _count = other count
+	}
+	copy: func -> This {
+		result := This new(this _count)
+		for (i in 0 .. this _count)
+			result add(this[i])
+		result
+	}
+	operator [] <T> (index: Int) -> T {
+		this as VectorList<Int> _vector[index]
+	}
+	operator []= (index: Int, item: Int) {
+		this _vector[index] = item
+	}
+	sort: func -> This {
+		result := this copy()
+		inOrder := false
+		while (!inOrder) {
+			inOrder = true
+			for (i in 0 .. result count - 1)
+				if (result[i] > result[i + 1]) {
+					inOrder = false
+					temporary := result[i]
+					result[i] = result[i + 1]
+					result[i + 1] = temporary
+				}
+		}
+		result
+	}
+	compress: func -> This {
+	// Returns a new, sorted list of unique elements, e.g. [1,2,1,5,2,7,6,7,7] -> [1,2,5,6,7]
```

Comment:
Where else does a function start with a comment (except for in the SDK, which I take 0 responsibility for)?